src/ucp/api/ucp.h

```diff
@@ -2177,8 +2177,105 @@ ucs_status_t ucp_worker_fence(ucp_worker_h worker);
  */
 ucs_status_t ucp_worker_flush(ucp_worker_h worker);
 
+/**
+ * @ingroup UCP_COMM
+ * @brief Atomic operation requested for ucp_atomic_post
+ *
+ * This enumeration defines which atomic memory operation should be performed
+ * by the ucp_atomic_post family of fuctions. All of these are send only
+ * and will not give back a request handle.
+ */
+typedef enum {
+    UCP_ATOMIC_POST_OP_ADD, /* Atomic add */
+    UCP_ATOMIC_POST_OP_LAST,
+} ucp_atomic_post_op_t;
+
+/**
+ * @ingroup UCP_COMM
+ * @brief Atomic operation requested for ucp_atomic_fetch
+ *
+ * This enumeration defines which atomic memory operation should be performed
+ * by the ucp_atomic_fetch family of functions. All of these functions
+ * will fetch data from the remote node
+ */
+typedef enum {
+    UCP_ATOMIC_FETCH_OP_FADD, /* Atomic Fetch and add */
+    UCP_ATOMIC_FETCH_OP_SWAP, /* Atomic swap */
+    UCP_ATOMIC_FETCH_OP_CSWAP, /* Atomic conditional swap */
+    UCP_ATOMIC_FETCH_OP_LAST,
+} ucp_atomic_fetch_op_t;
 
 /**
+ * @ingroup UCP_COMM
+ * @brief Post an atomic memory operation.
+ *
+ * This routine will post an atomic memory operation to remote memory.
+ * Return from the function does not guarantee any completion. A user must
```

Comment:
I meant that in terms of local or remote completion. I'll tighten this up. It is in the old blocking routines, I'll import that definition.